storage/2020-08-04/file/files/copy_and_wait_poller.go

```diff
@@ -0,0 +1,48 @@
+package files
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/sdk/client/pollers"
+)
+
+var _ pollers.PollerType = &copyAndWaitPoller{}
+
+func NewCopyAndWaitPoller(client *Client, shareName, path, fileName string) *copyAndWaitPoller {
+	return &copyAndWaitPoller{
+		client:    client,
+		shareName: shareName,
+		path:      path,
+		fileName:  fileName,
+	}
+}
+
+type copyAndWaitPoller struct {
+	client    *Client
+	shareName string
+	path      string
+	fileName  string
+}
+
+func (p *copyAndWaitPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	props, err := p.client.GetProperties(ctx, p.shareName, p.path, p.fileName)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving copy (shareName: %s path: %s fileName: %s) : %+v", p.shareName, p.path, p.fileName, err)
+	}
+
+	if strings.EqualFold(props.CopyStatus, "success") {
```

Comment:
(same here) can we use the constant:

```suggestion
	if strings.EqualFold(props.CopyStatus, string(Success)) {
```

Reply:
we don't have a constant for this in files atm, just blobs,  is it worth adding for this?

Reply:
Oh ok, I figured this was still around/used - in which case this is fine 👍